Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -55,6 +55,15 @@ def fix_dynamic_linkage
           new_name = fixed_name(file, bad_name)
           change_install_name(bad_name, new_name, file) unless new_name == bad_name
         end
+
+        each_linkage_for(file, :rpaths) do |bad_name|
+          # Strip rpaths rooted in the build directory
+          next if bad_name.start_with?("/") &&
+                  !bad_name.start_with?(HOMEBREW_TEMP.to_s) &&
+                  !bad_name.start_with?(HOMEBREW_TEMP.realpath.to_s)
```

Comment:
Doesn't this only allow absolute paths? Won't `@loader_path` be stripped?

Reply:
Nope, `each_linkage_for` ignores `@(loader_|executable_|r)path`:

https://github.com/Homebrew/brew/blob/7c68b1738b3dce2885d0146f327eaaf96b6d0029/Library/Homebrew/extend/os/mac/keg_relocate.rb#L86-L91

Reply:
I guess I'd also be stripping relative paths that don't start with `@(loader_|executable_|r)path`.

Reply:
Tbh it seems a little unsafe if Apple were to add `@new_path` tomorrow.

Reply:
`LC_RPATH` can be relative. But you're right, let's tighten this to just checking for the build directory.